src/geometries/ParametricGeometry.js

```diff
@@ -72,6 +77,16 @@ function ParametricBufferGeometry( func, slices, stacks ) {
 			var p = func( u, v );
 			vertices.push( p.x, p.y, p.z );
 
+			// approximate tangent plane vectors via central difference
+
+			pu.subVectors( func( u + EPS, v ), func( u - EPS, v ) );
```

Comment:
That's a good point! According to the implementation of `func`, values outside [ 0, 1 ] can be problematic.